prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -0,0 +1,298 @@
+    :param int veh_range: 100, 200, or 300, represents how far vehicle can travel on single charge.
+    :param int kwhmi: fuel efficiency, should vary based on vehicle type and model_year.
+    :param int power: charger power, EVSE kW.
+    :param int location_strategy: where the vehicle can charge-1, 2, 3, 4, or 5; 
+        1-home only, 2-home and work related, 3-anywhere if possibile, 
+        4-home and school only, 5-home and work and school.
+    :param str veh_type: determine which category (LDV or LDT) to produce charging profiles for
+    :return: (*numpy.ndarray*) -- charging profiles.
+    """
+
+    # load NHTS data from function
+    if veh_type.lower() == "ldv":
+        newdata = mileage.remove_ldt(mileage.load_data(census_region))
+
+    elif veh_type.lower() == "ldt":
+        newdata = mileage.remove_ldv(mileage.load_data(census_region))
+
+    # updates the weekend and weekday values in the nhts data
+    newdata = mileage.update_if_weekend(newdata)
+
+    # add new columns to newdata to store data that is not in NHTS data
+    newdata = newdata.reindex(
+        list(newdata.columns)
+        + [
+            "trip start battery charge",
+            "trip end battery charge",
+            "charging power",
+            "charging time",
+            "charging consumption",
+            "BEV could be used",
+            "trip number",
+        ],
+        axis="columns",
+        fill_value=0,
+    )
+
+    input_day = mileage.get_input_day(mileage.get_model_year_dti(model_year))
+
+    TRANS_charge = np.zeros(24 * len(input_day))
+    data_day = mileage.get_data_day(newdata)
+
+    trip_strategy = 1
+    kwh = kwhmi * veh_range
+    weekday_electricload, weekend_electricload = np.zeros(4800), np.zeros(4800)
+
+    l = len(newdata)
+
+    # weekday first
+    trip_num = 1
+    start_point = 0
+    end_point = 0
+
+    for i in range(l):
+        if data_day[i] == 2:  # 2 is weekday
+            if (
+                i > 0
+                and newdata.iloc[i, newdata.columns.get_loc("sample vehicle number")]
+                == newdata.iloc[i - 1, newdata.columns.get_loc("sample vehicle number")]
+            ):
+                newdata.iloc[
+                    i, newdata.columns.get_loc("trip start battery charge")
+                ] = (
+                    newdata.iloc[
+                        i - 1, newdata.columns.get_loc("trip end battery charge")
+                    ]
+                    + newdata.iloc[
+                        i - 1, newdata.columns.get_loc("charging consumption")
+                    ]
+                )
+                trip_num += 1
+                newdata.iloc[i, newdata.columns.get_loc("trip number")] = trip_num
+
+            else:
+                newdata.iloc[
+                    i, newdata.columns.get_loc("trip start battery charge")
+                ] = kwh
+                trip_num = 1
+                newdata.iloc[i, newdata.columns.get_loc("trip number")] = trip_num
```

Comment:
This code looks like we're grouping the trips by sample vehicle number, and adding values for `"trip number"` and `"trip start battery charge"`. I think we can do this in a clearer way by defining a function that calculates these values for all trips of a given vehicle at once, like we did with the `calculate_dwell_time` function within #243.